### PR TITLE
Wall Running

### PR DESCRIPTION
**Pull request by Geevies**
* Sprinting will now only drain your resources and deal damage if you managed to successfully move. The only exception is the calculation for when you're out of breath, as it will still make you hurt.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 1]
```diff
@@ -529,7 +529,8 @@
 		prescriptions -= H.equipment_prescription
 	return Clamp(prescriptions, 0, 7)
 
-/datum/species/proc/handle_sprint_cost(var/mob/living/carbon/human/H, var/cost)
+// if calculation is true, it won't sap resources
+/datum/species/proc/handle_sprint_cost(var/mob/living/carbon/human/H, var/cost, var/calculation, var/post_calculation)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/human/species/species.dm:533]: These variable names are the most undescriptive thing ever. Please find some better names-

Comment (Arrow768) [reply code/modules/mob/living/carbon/human/species/species.dm:533]: It might be worth to add a doc-block at the top aswell

**Review by JohnWildkins — APPROVED** [review]
Nice, no longer can you run out of stamina by "running" in a camera monitor.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**